source/draw/Canvas.ooc

```diff
@@ -67,6 +67,30 @@ Canvas: abstract class {
 	}
 	fill: abstract func (color: ColorRgba)
 	draw: virtual func ~DrawState (drawState: DrawState) { Debug error("draw~DrawState unimplemented for class " + this class name + "!") }
+	write: virtual func (target, fontAtlas: Image, localOrigin: IntPoint2D, message: Text) {
+		// Generic reference implementation with all inputs that are used from DrawState to simplify input space partitioning
+		skippedRows := 2 // Skip invisible commands
+		visibleRows := 6 // Use 6 lines of printable characters
+		columns := 16 // Use 16 characters per row
```

Comment:
That could indicate that the output uses word wrapping at 16 characters if there is no reference to a 2D atlas.